### PR TITLE
[SYCL-MLIR]: Update contributing instructions.

### PR DESCRIPTION
**Pull request by etiotto**
This PR outlines contributing instructions for the sycl-mlir branch.

### CODE REVIEW

**Comment by bader** [comment]
Should we remove https://github.com/intel/llvm/pull/6739/commits/39c32e1d2f6d85a935176c5aa0c58f52b39e2531 from this PR?

**Comment by pvchupin** [comment]
It probably makes sense to update CONTRIBUTING.md and README.md in default branch (sycl) instead.
That's what users see when open https://github.com/intel/llvm

Today these docs in default branch cover all common generic guidelines, license, etc. as well as listing all projects in this repo. 

And project specific contribution guidelines are in separate documentation somewhere in the project branch. See: https://github.com/intel/llvm/blob/sycl/sycl/doc/developer/ContributeToDPCPP.md for DPC++

**Comment by etiotto** [comment]
> It probably makes sense to update CONTRIBUTING.md and README.md in default branch (sycl) instead. That's what users see when open https://github.com/intel/llvm
> 
> Today these docs in default branch cover all common generic guidelines, license, etc. as well as listing all projects in this repo.
> 
> And project specific contribution guidelines are in separate documentation somewhere in the project branch. See: https://github.com/intel/llvm/blob/sycl/sycl/doc/developer/ContributeToDPCPP.md for DPC++
> 
> Basically I recommend:
> 
> * Add new ContributeToSYCLMLIR.md somewhere in sycl-mlir branch
> * Update README.md in sycl branch adding sycl-mlir project details.
> * Update CONTRIBUTING.md in sycl branch adding link to ContributeToSYCLMLIR.md

I saw that the openmp branch has its own readme: https://github.com/intel/llvm/tree/openmp so I modeled this PR after that branch. I can do it the way you recommend.

**Comment by etiotto** [comment]
Added ContributeToSYCLMLIR.md as suggested by @pvchupin

File: mlir-sycl/doc/developer/ContributeToSYCLMLIR.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+# Contributing to SYCL-MLIR
+
+## 1) Fork the Intel LLVM repository
+
+We recommend contributors to work in their own forks of the Intel LLVM project. To create a fork:
+	
+  1) Visit https://github.com/intel/llvm and click the Fork button to establish a fork (e.g. intel-llvm)
+	2) Go to your newly created fork, and get the URL of you fork (green Code button):
```
Comment (pvchupin) [inline mlir-sycl/doc/developer/ContributeToSYCLMLIR.md:8]: nit: list is not visualized properly.

File: mlir-sycl/doc/developer/ContributeToSYCLMLIR.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+```sh
+git checkout sycl-mlir
+git checkout -b <name-of-your-branch> 
+```
+
+Configure and build the project:
+
+```sh
+python3 ./buildbot/configure.py --build-type Debug   
+python3 ./buildbot/compile.py
```
Comment (pvchupin) [inline mlir-sycl/doc/developer/ContributeToSYCLMLIR.md:73]: nit: I thought just "python" should work here, should it be really python3 ?

File: mlir-sycl/doc/developer/ContributeToSYCLMLIR.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+```sh
+ninja check-mlir-sycl 
+ninja check-cgeist
+ninja check polygeist-opt
+```
+
+## 5) Commit changes
+
+At this point you are ready to commit and push your changes. Ensure you sign your commits (git commit -s) before pushing them, also please prepend commit messages with `[SYCL-MLIR]` for easy identification:
```
Comment (pvchupin) [inline mlir-sycl/doc/developer/ContributeToSYCLMLIR.md:86]: nit: We don't require signature in sycl branch (upstream) today. I would say it's optional.

**Review by pvchupin — APPROVED** [review]
LGTM. A few nits inlined.

File: mlir-sycl/doc/developer/ContributeToSYCLMLIR.md (hunk 1 above)
Comment (etiotto) [reply mlir-sycl/doc/developer/ContributeToSYCLMLIR.md]: done

File: mlir-sycl/doc/developer/ContributeToSYCLMLIR.md (hunk 2 above)
Comment (etiotto) [reply mlir-sycl/doc/developer/ContributeToSYCLMLIR.md]: Changed

**Review by whitneywhtsang — APPROVED** [review]

**Pull request merged**